error.go

```diff
@@ -187,10 +230,19 @@ type errProvide struct {
 	Reason error
 }
 
-func (e errProvide) cause() error { return e.Reason }
+var _ causer = errProvide{}
```

Comment:
I'm not familiar with the reasoning behind the pattern of var _ causer

Reply:
It's a compile-time check to verify that the interface is satisfied.

If errProvide fails to satisfy the causer interface, this code won't compile.

(We use this convention in a number of places.)

Reply:
👍